src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -367,8 +401,25 @@ public static bool TryText(FormattingInfo formatInfo, IRContext irContext, Formu
                         return TryExpandDateTimeExcelFormatSpecifiersToStringValue(irContext, formatString, "g", newDateTime.GetConvertedValue(timeZoneInfo), timeZoneInfo, culture, formatInfo.CancellationToken, out result);
                     }
                     else
-                    {
-                        result = new StringValue(irContext, num.Value.ToString(formatString ?? "g", culture));
+                    {
+                        if (!string.IsNullOrEmpty(formatString))
+                        {
+                            if (hasFormatCulture)
+                            {                                
+                                if (revFormatStr != formatString
+                                    && (num.Value.ToString(formatString, formatCulture) != num.Value.ToString(formatString, CultureInfo.GetCultureInfo(enUsCultureName))
+                                        || num.Value.ToString(revFormatStr, formatCulture) != num.Value.ToString(revFormatStr, CultureInfo.GetCultureInfo(enUsCultureName))))
+                                {
+                                    formatString = revFormatStr;
+                                }

```

Comment:
what is this doing?

Reply:
I explained above:

format string in c# only work for en-US format. They don't understand the , and . in different culture.
As in https://learn.microsoft.com/en-us/dotnet/standard/base-types/custom-numeric-format-strings.
So my idea here is if we pass different culture format, we finding the reverse version of it (will be in en-US format if the input is non en-US and have different format like fr-fr or vi-vi).
Then we use both format, the input string format and reverse input string format to compare the results of output string when we applied 2 different cultures we have here.
If any of comparison return different mean 2 cultures have different numeric format. Then we need to use reverse format (english format) first then applied that different culture (fr-fr for example).